cmake/catkin_libraries.cmake

```
@@ -114,6 +114,40 @@ function(catkin_unpack_libraries_with_build_configuration VAR)
   set(${VAR} "${result}" PARENT_SCOPE)
 endfunction()
 
+#
+# Find libraries by imported location or soname with location dir as hint
+#
+# :param VAR: the output variable name
+# :type VAR: string
+# :param LIB: the imported target to find the library for
+# :type LIB: string
+# :type ARGN: selected config to query in the target's properties
+# :type ARGN: string
+#
+# @public
+#
+function(catkin_find_library_imported_location_library VAR LIB)
+  set(lib ${LIB})
+  if(ARGN)
+    set(cfg _${ARGN})
+  endif()
+  set(imported_location_libnames)
+  get_target_property(imported_soname ${lib} IMPORTED_SONAME${cfg})
+  if(imported_soname)
+    list(APPEND imported_location_libnames ${imported_soname})
+  endif()
+  get_target_property(imported_location ${lib} IMPORTED_LOCATION${cfg})
+  if(imported_location)
+    get_filename_component(imported_location_name ${imported_location} NAME)
+    get_filename_component(imported_location_dir ${imported_location} DIRECTORY)
+    list(APPEND imported_location_libnames ${imported_location_name})
+  endif()
+  if(imported_location_libnames)
+    find_library(imported_location_library NAMES ${imported_location_libnames} HINTS ${imported_location_dir})
```

```suggestion
    find_library(imported_location_library NAMES ${imported_location_libnames} HINTS ${imported_location_dir} NO_CACHE)
```
Otherwise the first found library will be returned for all calls of this function. See https://github.com/RoboStack/ros-noetic/pull/426 .

---

I accepted it, but I did not follow up anymore it. So have not retested my use cases or if there is any situation in which cache needs to be used.